providers/in-process-flagd/README.md

```diff
@@ -0,0 +1,69 @@
+# In-process Flagd Provider
+
+This provider implements parts of the functionality of [Flagd](https://github.com/open-feature/flagd), and follows its
+specification for [in-process providers](https://github.com/open-feature/flagd/blob/main/docs/other_resources/in-process-providers/specification.md).
+
+## Setup
+To use flagd with the [OpenFeature SDK](https://github.com/open-feature/go-sdk) set the provider to the `openfeature` global singleton as shown below (using default values which align with those of `flagd`)
+```go
+openfeature.SetProvider(inprocessflagd.NewProvider())
+```  
+You may also provide additional options to configure the provider client
+```go
+inprocessflagd.WithHost(string)                  // defaults to localhost
+inprocessflagd.WithPort(uint16)                  // defaults to 8013
+inprocessflagd.FromEnv()                         // sets the provider configuration from environment variables
+inprocessflagd.WithSocketPath(string)            // no default, when set a unix socket connection is used (only available for GRPC)
+inprocessflagd.WithTLS(certPath string)          // default of false, if certPath is not given, system certs are used
+inprocessflagd.WithoutCache()                    // disables caching of flag evaluations
+inprocessflagd.WithLRUCache(1000)                // enables LRU caching (see configuring caching section)
+inprocessflagd.WithBasicInMemoryCache()          // enables basic in memory cache (see configuring caching section)
```

Comment:
I don't think these caching options are relevant to in-process providers.

Reply:
yeah, this should nearly match with Java provider configurations - https://github.com/open-feature/java-sdk-contrib/tree/main/providers/flagd#configuration-options